File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/card/application/query/GetCardResult.java
package com.example.accountservice.card.application.query;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDateTime;

public record GetCardResult(
        @Schema(description = "The card ID.") String cardId,
        @Schema(description = "The accountId this card is linked to.") String accountId,
        @Schema(description = "The userId of the card's owner.") String ownerId,
        @Schema(description = "The card network/brand.", example = "VISA") String brand,
        @Schema(description = "The card's status.", example = "ACTIVE") String status,
        @Schema(description = "When the card was issued.") LocalDateTime createdAt) {}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/card/application/query/GetCardResult.kt
package com.example.accountservice.card.application.query

import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

data class GetCardResult(
    @field:Schema(description = "The card's ID (32-character hex, no hyphens).", example = "d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1")
    val cardId: String,
    @field:Schema(description = "The account this card is linked to.", example = "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4")
    val accountId: String,
    @field:Schema(description = "The userId of the card owner.", example = "jane.doe")
    val ownerId: String,
    @field:Schema(description = "The card network brand.", example = "VISA")
    val brand: String,
    @field:Schema(description = "The card's lifecycle status.", example = "ACTIVE")
    val status: String,
    @field:Schema(description = "When the card was issued.")
    val createdAt: LocalDateTime,
)
